File: typescript/medium/getAverages.ts
function getAverages(nums: number[], k: number): number[] {
  let cur: number = 0;
  const l: number = 2 * k + 1;
  const avg: number[] = new Array<number>(nums.length).fill(-1);
  let left: number = 0;
  let right: number = 0;
  while (right < l) {
    cur += nums[right++];
  }
  avg[right - k - 1] = Math.floor(cur / l);
  while (right < nums.length) {
    cur += nums[right++];
    cur -= nums[left++];
    avg[right - k - 1] = Math.floor(cur / l);
  }
  return avg;
}

File: java/src/medium/GetAverages.java
package medium;

import java.util.Arrays;

public class GetAverages {
    public int[] getAverages(int[] nums, int k) {
        int[] res = new int[nums.length];
        Arrays.fill(res, -1);
        int len=2*k+1;
        if(nums.length<len){
            return  res;
        }
        long cur=0;
        int left=0;
        int right=0;
        while(right<len){
            cur+=nums[right++];
        }
        res[right-k-1]=(int)cur/len;
        while(right<nums.length){
            cur+=nums[right++];
            cur-=nums[left++];
            res[right-k-1]=(int)cur/len;
        }

        return res;
    }
}
